code/ui/panelcontent/PanelContent.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+
+using Sandbox.UI;
+
+namespace TTTReborn.UI
+{
+    public struct PanelContentData
+    {
+        public Action<PanelContent> Content;
+        public string Title;
+        public string ClassName;
+
+        public PanelContentData(Action<PanelContent> content, string title = "", string className = "")
+        {
+            Content = content;
+            Title = title;
+            ClassName = className;
+        }
+    }
+
+    public partial class PanelContent : TTTPanel
+    {
+        public string Title { get; private set; } = "";
+
+        public bool CanPrevious
+        {
+            get
+            {
+                return _historyIndex > 0;
+            }
+        }
+
+        public bool CanNext
+        {
+            get
+            {
+                return _historyIndex < (_contentHistory.Count - 1);
+            }
+        }
+
+        public Action<PanelContentData> OnPanelContentUpdated { get; set; }
+
+        public PanelContentData? CurrentPanelContentData { get; private set; }
+
+        private readonly List<PanelContentData> _contentHistory = new();
+
+        private int _historyIndex = 0;
+
+        public PanelContent(Panel parent = null) : base()
+        {
+            Parent = parent;
+        }
+
+        public void Reset()
+        {
+            _contentHistory.Clear();
+            _historyIndex = 0;
+
+            DeleteChildren(true);
+        }
+
+        public void SetPanelContent(Action<PanelContent> createContent, string title = "", string className = "")
+        {
+            int historyOffset = _contentHistory.Count - _historyIndex - 1;
+
+            if (historyOffset > 0)
+            {
+                _contentHistory.RemoveRange(_historyIndex + 1, historyOffset);
+            }
+
+            _contentHistory.Add(new PanelContentData(createContent, title, className));
+            _historyIndex = _contentHistory.Count - 1;
+
+            UpdatePanelContent();
+        }
+
+        public void Previous()
+        {
+            if (_historyIndex < 1)
+            {
+                return;
+            }
+
+            _historyIndex--;
+
+            UpdatePanelContent();
+        }
+
+        public void Next()
+        {
+            if (_historyIndex > _contentHistory.Count - 2)
```

Comment:
`-2` seems a bit odd here, is it because we need to subtract `-1 from both?

Reply:
Because we need to check that we are not at the last index (one before the last index, and as Counts return the index+1, we have to do `-2`)